ecl/hql/hqlgram2.cpp

```diff
@@ -4974,6 +4977,25 @@ void HqlGram::ensureUnicode(attribute &a)
     }
 }
 
+void HqlGram::ensureUTF8(attribute &a)
+{
+    ITypeInfo *t1 = a.queryExprType();
+    if (t1 && !isUTF8Type(t1))
+    {
+        if (isStringType(t1))
```

Comment:
minor: is StringType() || isUnicodeType() ?

Reply:
Good catch; changed.

isUnicodeType() tests for UTF-8 and there seems to be a lot of code that leverages the implied "UTF-8 is a subset of Unicode" logic.  This is debatable, but not something to fix here.